scripts/create_amy_epw_files.py

```diff
@@ -32,7 +32,7 @@ def clean_noaa_df(df):
                           'Liquid_Precipitation_Depth_Dimension_1H', 'Liquid_Precipitation_Depth_Dimension_6H'])
 
     # Create series of continuous timestamp values for that year
-    all_timestamps = pd.date_range(df['obs_timestamps'].iloc[0], periods=8760, freq='H')
+    all_timestamps = pd.date_range(df['obs_timestamps'].iloc[0], df['obs_timestamps'].iloc[-1], freq='H')
```

Comment:
This ensures that we cover the precise timespan covered by the input file. We could also keep the hard-coded number of periods but also hard-code the start time to 00:00 January 1; then we'd be sure to get every hour in the year, but we'd get an NA row for most years since entries for 00:00 on January 1 only appear in a small fraction of the data files. I'm not sure what suits our purposes better.

Reply:
This is a great catch, thank you. I think it makes sense to actually hard-code the start time to 00 January 1. That way we're capturing every hour in the year, as you say, and treating the missing hour as a missing data point. We will get a lot of NA rows in the first hour, but since that information isn't recorded in the file for the previous year, it is indeed missing. 

[I'm guessing that the reason these observations are often missing is that technically they would have been associated with the previous year, and their data management infrastructure might not be set up to handle that transition well. The measurements are recorded in a “capture window...defined as observations with observation minute greater than or equal to ten minutes before the top of the hour up to and including the top of the hour" per https://www1.ncdc.noaa.gov/pub/data/noaa/isd-lite/isd-lite-technical-document.pdf. So they probably only caught observations that would have been recorded with a timestamp at exactly the top of the hour.]